File: calificaciones.py
##Calificaciones.

estudiantes = []  
while True:   
    print("Materia")   
    nombre = input("Escribe tu nombre: ")   
    if nombre == "Salir":    
        break   
    estudianteExistente = next((est for est in estudiantes if est[0].lower() == nombre.lower()), None)
    
    if estudianteExistente:
        materias = estudianteExistente[1]
        print(f"El estudiante {nombre} ya está registrado.")
    else:
        materias = []
        estudiantes.append([nombre, materias]) 
    while True:    
        nombre_materia = input("Ingrese la materia: ")     
        if nombre_materia == "Salir":      
            break  
        nota = input("Ingrese su nota: ")     
        calificaciones = {nombre_materia:nota}    
        for idx, value in calificaciones.items():      
            print(idx,":", value)    
        
        materias.append(calificaciones)    
    print(f"Estudiante {nombre} registrado con éxito.\n")  
print(estudiantes)
    

File: calificaciones.js
// Calificaciones
let estudiantes = []
 while (true){
 console.log("Materias")
 let nombre = prompt("Ingrese su nombre: ")
 if (nombre == "Salir")
    {break}
 let estudianteExistente = estudiantes.find(est => est[0].toLowerCase() === nombre.toLowerCase())
 let materias = estudianteExistente ? estudianteExistente[1] : []; // Si existe, obtener sus materias

 if (estudianteExistente) {
     console.log(`El estudiante ${nombre} ya está registrado.`);
 }
 while(true){
    let nombreMateria = prompt("Ingrese la materia:")     
     if( nombreMateria == "Salir"){    
            break;
         }
     let nota = prompt("Ingrese su nota")  
     nota = parseFloat(nota);  
     let calificaciones = {}  
     calificaciones[nombreMateria] = nota;
     for (let [materia, calificacion] of Object.entries(calificaciones)) {
         console.log(materia, ":", calificacion);
     }
     if (estudianteExistente) {
      estudianteExistente[1] = materias;
  } else {
      
      estudiantes.push([nombre, materias]);
  }
     materias.push(calificaciones);  
  }

  estudiantes.push([nombre, materias]);  
  console.log(`Estudiante ${nombre} registrado con éxito.\n`);
 }


console.log(estudiantes);

 